### ISSUE TITLE
Support multiple webpack configurations

### ISSUE THREAD

**Issue by pyk** [cloudflare/wrangler#1397]
#### Overview and problem statement

It would be nice if wrangler support multiple webpack configurations.

The use case is when the user need to bundle static assets that runs in the browser.

Enabling this feature would greatly improve the developer experiences when publishing or developing Workers Sites locally.

#### Basic example

For example:

```js
// Webpack config to bundle worker script
const webWorkerBundleConfig = {
    target: "webworker",
    // redacted
};

// Webpack config to bundle javascript assets that run in the browser
const javascriptBrowserBundleConfig = {
    target: "web",
    // redacted
};

module.exports = [javascriptBrowserBundleConfig, webWorkerBundleConfig];
```

#1047

**Comment by pyk** [comment]
My current approach is to use separate file for this:

1. `webpack.workers.js`: webpack configuration to bundle workers script
2. `webpack.sites.js`: webpack configuration to bundle static assets that run in the browser

On local development, I run the following command:

```
# terminal 1: to watch & bundle static asset based on changes
npx onchange 'src/pages/**/*.html' 'src/assets/**/*' -- npx webpack --config webpack.sites.js --mode=development

# terminal 2: to run workers sites locally
wrangler dev
```

To publish, I use the following command:

```
npx webpack --config webpack.sites.js --mode=production && wrangler publish -e production
```

**Comment by lizzo-tlg** [comment]
Up!

**Comment by xortive** [comment]
closing since this is addressed by #1677, which will be going out as part of an RC in the near future.
